src/components/TodoItem.tsx

```diff
@@ -21,8 +21,10 @@ export const TodoItem: FC<Props> = ({
   const handleClick = () => {
     setIsBeingDeleted(true);
 
-    deleteTodo(todo.id).catch(() => {
-      setIsBeingDeleted(false);
+    deleteTodo(todo.id).then(didSucceed => {
+      if (!didSucceed) {
+        setIsBeingDeleted(false);
+      }
```

Comment:
deleteTodo don't return promise so you can't use than, catch or finally (these methods are called on a promise to handle the resolution or rejection of the promise. If the deleteTodo function returns a value other than a promise, you won't be able to handle it using promise-related methods)